apimodels/agent_models.go

```diff
@@ -92,3 +107,63 @@ type EbsDevice struct {
 	SizeGiB    int    `mapstructure:"ebs_size" json:"ebs_size"`
 	SnapshotID string `mapstructure:"ebs_snapshot_id" json:"ebs_snapshot_id"`
 }
+
+func (ch *CreateHost) Validate() error {
```

Comment:
Should there be an additional validation step that one and only one of `UserdataFile` and `UserdataCommand` are set?